### PR TITLE
Add typed-continuations feature and cont type

### PR DESCRIPTION
**Pull request by frank-emrich**
This PR does two things:
1. Add a Wasm feature called `typed-continuations`
2. Extended the language of types supported with `cont` types.

Note that the very first commit in this PR deletes a few existing test files. These were failing on my system on a vanilla checkout of binaryen (even on the most recently released tag). I need to investigate this further before merging this PR.

### CODE REVIEW

File: src/wasm-binary.h [hunk 1]
```diff
@@ -424,6 +424,7 @@ enum EncodedType {
   stringview_iter = -0x1f,  // 0x61
   // type forms
   Func = -0x20,   // 0x60
+  Cont = -0x23,   // FIXME(frank-emrich) what do these comments mean?
```
Comment (dhil) [inline src/wasm-binary.h:427]: That's the signed LEB encoding. Type constructor opcodes are encoded using unsigned LEB. See https://webassembly.github.io/spec/core/binary/types.html

File: src/wasm-features.h [hunk 2]
```diff
@@ -44,11 +44,13 @@ struct FeatureSet {
     ExtendedConst = 1 << 13,
     Strings = 1 << 14,
     MultiMemory = 1 << 15,
+    // FIXME(frank-emrich) Model dependencies on ref types and exceptions?
+    TypedContinuations = 1 << 16,
```
Comment (dhil) [inline src/wasm-features.h:48]: I wonder whether we should change name to `wasmfx`... it is easier to type and catchier to pronounce (and more memorable!).

File: src/wasm/wasm-s-parser.cpp [hunk 3]
```diff
@@ -948,6 +976,11 @@ void SExpressionWasmBuilder::preParseHeapTypes(Element& module) {
       Element& subtypeKind = *subtype[0];
       if (subtypeKind == FUNC) {
         builder[index] = parseSignatureDef(subtype, 0);
+      } else if (kind == CONT) {
+        // FIXME(frank-emrich) Should we support this?
+        throw ParseException("continuation type in subtype definition",
+                             subtypeKind.line,
+                             subtypeKind.col);
```
Comment (dhil) [inline src/wasm/wasm-s-parser.cpp:983]: For now I think it is fine to error. We want to have the proper design in place in the reference interpreter with the GC extension before we start implementing anything in the toolchains.

File: src/wasm-binary.h (hunk 1 above)
Comment (dhil) [reply src/wasm-binary.h:427]: I chose `-0x23` arbitrarily. The signed encoding is `5d`, I believe.

**Comment by dhil** [comment]
Should we restore the deleted files before merging this?

**Comment by dhil** [comment]
@frank-emrich can you please rebase before merge, and also have you had a look at restoring the deleted tests?

**Comment by dhil** [comment]
@frank-emrich can we close this PR?

**Comment by frank-emrich** [comment]
yes

**Pull request closed without merging**